pkg/build/reaper/reaper_test.go

```diff
@@ -0,0 +1,195 @@
+package reaper
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kerrors "k8s.io/kubernetes/pkg/api/errors"
+	ktestclient "k8s.io/kubernetes/pkg/client/unversioned/testclient"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	"github.com/openshift/origin/pkg/client/testclient"
+)
+
+func mkbuildconfig(version int) *buildapi.BuildConfig {
+	return &buildapi.BuildConfig{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:      "config",
+			Namespace: "default",
+		},
+		Spec: buildapi.BuildConfigSpec{},
+		Status: buildapi.BuildConfigStatus{
+			LastVersion: version,
+		},
+	}
+}
+
+func mkbuild(version int) buildapi.Build {
+	return buildapi.Build{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:      fmt.Sprintf("build-%d", version),
+			Namespace: "default",
+			Labels:    map[string]string{buildapi.BuildConfigLabel: "config"},
+		},
+	}
+}
+
+func mkdeprecatedbuild(version int) buildapi.Build {
+	return buildapi.Build{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:      fmt.Sprintf("build-%d", version),
+			Namespace: "default",
+			Labels:    map[string]string{buildapi.BuildConfigLabelDeprecated: "config"},
+		},
+	}
+}
+
+func mkbuildlist(version int) *buildapi.BuildList {
+	if version%2 != 0 {
+		panic("version needs be even")
+	}
+	list := &buildapi.BuildList{}
+	for i := 1; i <= version; i += 2 {
+		list.Items = append(list.Items, mkbuild(i))
+		list.Items = append(list.Items, mkdeprecatedbuild(i+1))
+	}
+	return list
+}
+
+func newBuildListFake(objects ...runtime.Object) *testclient.Fake {
+	fake := testclient.NewSimpleFake(objects...)
+	fake.PrependReactor("list", "builds", func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
+		selector := action.(ktestclient.ListAction).GetListRestrictions().Labels
+		retList := &buildapi.BuildList{}
+		for _, obj := range objects {
+			list, ok := obj.(*buildapi.BuildList)
+			if !ok {
+				continue
+			}
+			for _, build := range list.Items {
+				if selector.Matches(labels.Set(build.Labels)) {
+					retList.Items = append(retList.Items, build)
+				}
+			}
+		}
+		return true, retList, nil
+	})
+	return fake
+}
+
+func TestStop(t *testing.T) {
+	notfound := func() runtime.Object {
+		return &(kerrors.NewNotFound("BuildConfig", "config").(*kerrors.StatusError).ErrStatus)
+	}
+
+	tests := []struct {
+		testName  string
+		namespace string
+		name      string
+		oc        *testclient.Fake
+		expected  []ktestclient.Action
+		output    string
+		err       bool
+	}{
```

Comment:
> Instead of embedding a testName into struct do `map[string]struct{}`.

I find it a bit less convenient to use this type of more evolved table-driven test with a map.
Maps have no guaranteed order; don't allow us to print a consistent index to the test case; make it impossible to, while debugging, run only a fraction of the tests by slicing the test cases, ...

And I haven't seen people using maps for that purpose in other projects in the Go community.

On the plus side, maps wouldn't allow you to accidentally give the same name to two or more test cases.

There are no occurrences of `map[string]struct` in Go's standard library, while there are some table-driven tests using a name field to describe the test.

- https://github.com/golang/go/blob/86f1944e86ae3e07651dcbfafee37f1e3b0a917d/src/cmd/api/goapi_test.go#L82
- https://github.com/golang/go/blob/86f1944e86ae3e07651dcbfafee37f1e3b0a917d/src/crypto/cipher/cbc_aes_test.go#L29

@soltysh why are you advocating using maps?